x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -54,273 +56,335 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
+          } as unknown as NotificationRuleTypeParams;
+
+          // check if rule has permissions to access given index pattern
+          // move this collection of lines into a function in utils
+          // so that we can use it in create rules route, bulk, etc.
+          try {
+            // Typescript 4.1.3 can't figure out that `!isMachineLearningParams(params)` also excludes the only rule type
+            // of rule params that doesn't include `params.index`, but Typescript 4.3.5 does compute the stricter type correctly.
+            // When we update Typescript to >= 4.3.5, we can replace this logic with `!isMachineLearningParams(params)` again.
+            if (
+              isEqlParams(params) ||
+              isThresholdParams(params) ||
+              isQueryParams(params) ||
+              isSavedQueryParams(params) ||
+              isThreatParams(params)
+            ) {
+              const index = params.index;
+              const hasTimestampOverride = !!timestampOverride;
+
+              const inputIndices = params.index ?? [];
+
+              const privileges = await checkPrivilegesFromEsClient(esClient, inputIndices);
+
+              wroteWarningStatus = await hasReadIndexPrivileges({
                 ...basicLogArguments,
-                timestampField: hasTimestampOverride ? (timestampOverride as string) : '@timestamp',
-                timestampFieldCapsResponse: timestampFieldCaps,
-                inputIndices,
-                ruleStatusClient,
+                privileges,
                 logger,
                 buildRuleMessage,
+                ruleStatusClient,
               });
+
+              if (!wroteWarningStatus) {
+                const timestampFieldCaps = await withSecuritySpan('fieldCaps', () =>
+                  services.scopedClusterClient.asCurrentUser.fieldCaps({
+                    index,
+                    fields: hasTimestampOverride
+                      ? ['@timestamp', timestampOverride as string]
+                      : ['@timestamp'],
+                    include_unmapped: true,
+                  })
+                );
+                wroteWarningStatus = await hasTimestampFields({
+                  ...basicLogArguments,
+                  timestampField: hasTimestampOverride
+                    ? (timestampOverride as string)
+                    : '@timestamp',
+                  timestampFieldCapsResponse: timestampFieldCaps,
+                  inputIndices,
+                  ruleStatusClient,
+                  logger,
+                  buildRuleMessage,
+                });
+              }
             }
+          } catch (exc) {
+            const errorMessage = buildRuleMessage(`Check privileges failed to execute ${exc}`);
+            logger.error(errorMessage);
+            await ruleStatusClient.logStatusChange({
+              ...basicLogArguments,
+              message: errorMessage,
+              newStatus: RuleExecutionStatus['partial failure'],
```

Comment:
Should we be setting `agent.setTransactionOutcome('failure');` here (and in other failure/success cases) just as you did over in `signal_rule_alert_type`?

Reply:
Ahhh, is this covered globally by `task_runner`?

https://github.com/elastic/kibana/blob/1b027422436cf9e99e75f713ac2fd1db666f9ef6/x-pack/plugins/alerting/server/task_runner/task_runner.ts#L640-L646

Reply:
Yes, they added it recently, so there's no need to set the outcome on our side anymore. Thanks for pointing it out. I also removed `setOutcome` from `signal_rule_alert_type `.